ouroboros-consensus/test-consensus/Test/Consensus/Mempool.hs

```diff
@@ -124,6 +128,127 @@ prop_Mempool_InvalidTxsNeverAdded setup =
         , txInMempool `notElem` txsInMempoolBefore
         ]
 
+-- | When the mempool is at capacity, test that 'addTxs' blocks when
+-- attempting to add one more transaction and that it unblocks when there is
+-- adequate space. Adequate space is made by adding all of the existing
+-- transactions to the ledger and removing them from the mempool.
+prop_Mempool_Capacity :: Property
+prop_Mempool_Capacity =
+  let nbInitialTxs = 0
+  in  forAll (choose (2, 100)) $ \nbNewTxs ->
+      forAll (genTestSetup nbInitialTxs (nbNewTxs - 1)) $ \(setup@TestSetup{testLedgerState}, _) ->
+      forAll (genValidTxs nbNewTxs testLedgerState) $ \(vtxs, _) ->
+        withTestMempool setup $ runCapacityTest (map TestGenTx vtxs)
+ where
```

Comment:
I'll have a look at these tests later, if you don't mind.

Reply:
No problem